### PR TITLE
Ravil/writers

### PR DESCRIPTION
**Pull request by ravil-mobile**
These changes are intended to fix result writers while computing with single precision. An important point that geometry is still handled with double precision but data can be written to hdf5 files with single precision fp. format. It led to some changes in XdmfWriter sub-module (changes can be seen in this [PR](https://github.com/TUM-I5/XdmfWriter/pull/1)). 

DR-single-output
WF-single-output

I've tested only TPV5 benchmark so far. However, a more detailed evaluation is required.

**Important:** this PR is not going to fix *check-pointing*. It will be done in a different PR. This PR is only focused on result writers

### CODE REVIEW

**Comment by sebwolf-de** [comment]
Hi, LGTM, but should we wait for the "more detailed evaluation" before merging?

**Comment by ravil-mobile** [comment]
Thanks @sebwolf-de. I am going to fix tests and test some other scenarios. I think It would be nice if @Thomas-Ulrich can check this out.

**Comment by uphoffc** [comment]
Don't think we need to wait on more detailed testing. While I'm not sure whether double -> real was substituted everywhere, but this PR doesn't break things for double precision

**Comment by Thomas-Ulrich** [comment]
Thx for this PR! 
I will test the patch in the next days.

**Comment by ravil-mobile** [comment]
Hi all, 

I fixed a unit test and everything is working now. I will focus on suggestions of @sebwolf-de and @uphoffc about XdmfWriter.

> Don't think we need to wait on more detailed testing. While I'm not sure whether double -> real was substituted everywhere, but this PR doesn't break things for double precision

agree

**Comment by Thomas-Ulrich** [comment]
Hi, the plasticity (eta, eps_xx) outputs are still not working in single precision.
Fault, surface, and wavefield output looks fine

**Comment by ravil-mobile** [comment]
Hi all, 

Here is just some info. I rebased this branch with the current master to keep the project history clean and (forced) pushed it here. So, all my changes are at the top of the thread pinning stuff.

**Comment by Thomas-Ulrich** [comment]
ok now it is working fine!

**Comment by ravil-mobile** [comment]
Thanks @Thomas-Ulrich!

Can somebody approve these changes?

File: src/ResultWriter/FaultWriterC.cpp
Comment (Thomas-Ulrich) [inline src/ResultWriter/FaultWriterC.cpp:46]: is it only formatting changes?
Better uncommit that?

File: src/ResultWriter/FaultWriterC.cpp
Comment (Thomas-Ulrich) [inline src/ResultWriter/FaultWriterC.cpp:58]: same

File: src/ResultWriter/FaultWriterExecutor.h
Comment (Thomas-Ulrich) [inline src/ResultWriter/FaultWriterExecutor.h:48]: same

File: src/ResultWriter/FaultWriterExecutor.h
Comment (Thomas-Ulrich) [inline src/ResultWriter/FaultWriterExecutor.h:50]: same

File: src/ResultWriter/FreeSurfaceWriterExecutor.cpp
Comment (Thomas-Ulrich) [inline src/ResultWriter/FreeSurfaceWriterExecutor.cpp:88]: same

File: src/ResultWriter/FaultWriterC.cpp
Comment (Thomas-Ulrich) [reply src/ResultWriter/FaultWriterC.cpp:46]: (sorry I know it is annoying)

Comment (ravil-mobile) [reply src/ResultWriter/FaultWriterC.cpp:46]: why? The old formatting was difficult to read.

Comment (Thomas-Ulrich) [reply src/ResultWriter/FaultWriterC.cpp:46]: because it is then difficult to track changes?
We could automate the formatting at some point as we did for PUMGEN?
I had a similar review in another PR.

Comment (Thomas-Ulrich) [reply src/ResultWriter/FaultWriterC.cpp:46]: anyway, that was just my 2 cents, you got my approval if you don't want to change it.

**Review by Thomas-Ulrich — APPROVED** [review]

File: src/ResultWriter/FaultWriterC.cpp
Comment (ravil-mobile) [reply src/ResultWriter/FaultWriterC.cpp:46]: >because it is then difficult to track changes?

Really?

>  We could automate the formatting at some point as we did for PUMGEN?

I also proposed clang-format for SeisSol a long ago but it was rejected (or it is better to say postponed). But it is inevitable and at tome point it is going to be introduced. I am personally for introducing clang-format to SeisSol as soon as possible. I know that it will introduce lot's of changes in one commit and lot's people will complain whose branches are a bit behind of the master. But on the other hand, there will never be perfect time to seamlessly introduce clang-format to SeisSol.

**Comment by Thomas-Ulrich** [comment]
fixes #80

**Pull request merged**
